src/core/drive/page_view.js

```diff
@@ -56,7 +56,7 @@ export class PageView extends View {
   }
 
   isPageRefresh(visit) {
-    return !visit || (this.lastRenderedLocation.href === visit.location.href && visit.action === "replace")
+    return !visit || (this.lastRenderedLocation.pathname === visit.location.pathname && visit.action === "replace")
```

Comment:
I'd check that the `host` matches too. It would be a weird scenario that it didn't but I think. You could extract a method `isReloadingCurrentPath` that checks both conditions: `pathname` and `host`.

Reply:
Great point @brunoprietog!